src/lang/translation_provider_FILE.hpp

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include "translation_provider.hpp"
+#include <stdio.h>
+#include <string.h>
+#include "gettext_string_hash.hpp"
+
+//TODO fix including from string.h
+extern "C" size_t strlcpy(char *dst, const char *src, size_t dsize);
+
+class FILEtranslationProvider : public ITranslationProvider {
+
+    uint16_t findMessage(const char *key, FILE *file) const {
+
+        uint32_t index = m_HashTable.GetIndexOfKey(key);
+
+        //nothing is on index 0
+        if (index == 0) {
+            return false;
+        }
+
+        //get to position of translated string
+        fseek(file, m_TransTableOff + (8 * index) + 4, SEEK_SET);
+
+        uint32_t posOfTrans = 0;
+        fread(&posOfTrans, 4, 1, file);
+        fseek(file, posOfTrans, SEEK_SET);
+        return posOfTrans;
+    }
+
+    char m_Path[16];
+    FILE *m_File = nullptr;
+    gettext_hash_table m_HashTable;
+    uint32_t m_TransTableOff;
+
+public:
+    FILEtranslationProvider(const char *path) {
+#ifdef TRANSLATIONS_UNITTEST
```

Comment:
this should't be necessary anymore, we have the strlcpy implementation for the unit tests as well now